### PR TITLE
restructure and reorganize CI scripts and documentation

### PR DESCRIPTION
**Pull request by eslutsky**
in this PR:
- moved  the scripts  into a logical directory structure from the project-root folder.
- add collapsible output to long  log workflow action output.
- OSP: update public endpoints to forklift url.
- OSP: add cinder port to the deployment
- update the documentation

### CODE REVIEW

File: build_and_setup_everything_bazel.sh [hunk 1]
```diff
@@ -1,56 +1,74 @@
...
 esac
+echo "::endgroup::"
+
+source ./cluster/common.sh 
+
+# grant admin rights so its token can be used to access the API
+k8s_grant_permissions
 
-. ./grant_permissions.sh
+# pactch StorageProfile with ReadWriteOnce Access
```
Comment (ahadas) [inline build_and_setup_everything_bazel.sh:70]:
```suggestion
# patch StorageProfile with ReadWriteOnce Access
```

File: cluster/providers/ovirt/setup.sh [hunk 2]
```diff
@@ -2,14 +2,14 @@
 
 set -ex
 
-kubectl apply -f ovirt/patch.yml
-kubectl apply -f ovirt/fakeovirt_deployment.yml
-kubectl apply -f ovirt/ovirt_imageio_deployment.yml
+kubectl apply -f cluster/providers/ovirt/patch.yml
```
Comment (ahadas) [inline cluster/providers/ovirt/setup.sh:5]: do we still need to do that? (and if not, can we drop patch.yml?)

**Review by ahadas — COMMENTED** [review]
I like this cleanup!
besides the comments inside, do we still need to keep the grant_permissions.sh script?

File: cluster/providers/ovirt/setup.sh (hunk 2 above)
Comment (eslutsky) [reply cluster/providers/ovirt/setup.sh]: yeap , it can be dropped.

File: cluster/providers/openstack/utils.sh [hunk 3]
```diff
@@ -128,13 +128,32 @@ function packstack_create_cirros_volume {
...
+
+function packstack_update_endpoints {
+source /root/keystonerc_admin
+
+openstack endpoint list --interface public  -c ID -c URL -f value | grep 127.0.0.1 >/tmp/output
+
+new_host="packstack.${namespace_name}"
+while IFS=" " read -r id url;     do 
+    openstack endpoint set --url ${url/127.0.0.1/"$new_host"} $id
+done  < /tmp/output
```
Comment (ahadas) [inline cluster/providers/openstack/utils.sh:157]: wrong indentation?

Comment (eslutsky) [reply cluster/providers/openstack/utils.sh]: fixed

**Review by ahadas — APPROVED** [review]
ok, let's drop grant_permissions.sh separately

File: BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md [hunk 4]
```diff
@@ -12,96 +12,37 @@ used as a VM migration target.
...
-[patch_for_local_registry.sh](patch_for_local_registry.sh) in the same directory where the
-forklift repositories were checked out.  
-*Warning:* the patch matches the state of the forklift repositories on
-2022-09-12. It might become obsolete of the files which need patching are
-updated.
+# providers
+we are installing those providers as part of k8s kind cluster :
+- [VMware - vcsim](https://github.com/vmware/govmomi/blob/main/vcsim/README.md)
+- [openstack - packstack](https://github.com/kubev2v/packstack-img)
+- [oVirt - fakeovirt](https://github.com/kubev2v/fakeovirt)
```
Comment (ahadas) [inline BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md:41]:
```suggestion
- [oVirt - fakeovirt and ovirt-imageio](https://github.com/kubev2v/fakeovirt)
```

File: BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md [hunk 5]
```diff
@@ -12,96 +12,37 @@ used as a VM migration target.
 Only the backend is installed, i.e. operator/controller/validation, without
 the UI, because it is meant to be the target for running automated tests.
 
-
-# Prerequisits
-
+# Pre-requisite
```
Comment (ahadas) [inline BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md:15]: `Prerequisits` is ok, no need to change it (see [our doc](https://access.redhat.com/documentation/en-us/migration_toolkit_for_virtualization/2.3/html-single/installing_and_using_the_migration_toolkit_for_virtualization/index#prerequisites))

File: build_and_setup_everything_bazel_manually.sh [hunk 6]
```diff
@@ -1,29 +1,46 @@
...
 
-./ovirt/setup.sh
+# grant admin rights so its token can be used to access the API
+k8s_grant_permissions
 
-./openstack/setup.sh
+# patch StorageProfile with ReadWriteOnce Access
+k8s_patch_storage_profile
 
-. ./grant_permissions.sh
```
Comment (ahadas) [inline build_and_setup_everything_bazel_manually.sh:26]: so, can we drop that script from the repo then? :)

File: BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md (hunk 5 above)
Comment (eslutsky) [reply BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md:15]: its Prerequisites :)

File: build_and_setup_everything_bazel_manually.sh (hunk 6 above)
Comment (eslutsky) [reply build_and_setup_everything_bazel_manually.sh:26]: dropps

**Review by ahadas — APPROVED** [review]

**Pull request merged**